tools/PI/DevHome.PI/Helpers/CommonHelper.cs

```diff
@@ -141,5 +140,32 @@ public static string GetCommandLine(Process process)
         }
 
         return activationArgs;
-    }
+    }
+
+    private static readonly Int32Converter _converter = new();
+
+    public static int? ParseStringToInt(string value)
+    {
+        int? valueAsInt;
+
+        try
+        {
+            if (_converter.IsValid(value))

```

Comment:
https://learn.microsoft.com/dotnet/csharp/programming-guide/types/how-to-convert-between-hexadecimal-strings-and-numeric-types shows that the Int32 parser can parse hex numbers without the leading 0x.

Reply:
Yep, and to do that you need to pass in the NumberStyles.HexNumber flag.

It makes sense that this is difficult when you don't know if strings contain hexadecimal or decimal numbers.... it would be nice to have a function treat abc and 0xabc the same, but what about 123 and 0x123?